packages/runtime/container-runtime/src/pendingStateManager.ts

```diff
@@ -80,7 +95,34 @@ export class PendingStateManager {
         return this.pendingMessagesCount !== 0;
     }
 
-    constructor(private readonly containerRuntime: ContainerRuntime) { }
+    public getLocalState(): IPendingLocalState | undefined {
+        if (this.hasPendingMessages()) {
+            return {
+                clientId: this.clientId,
+                pendingStates: this.pendingStates.toArray(),
+            };
+        }
+    }
+
+    constructor(
+        private readonly containerRuntime: ContainerRuntime,
+        private readonly rebaseOp: (content, localOpMetadata) => Promise<void>,
+        initialState: IPendingLocalState | undefined,
+    ) {
+        this.initialStates = new Deque<IPendingState>(initialState?.pendingStates ?? []);
+        this.initialClientId = initialState?.clientId;
+
+        // get client sequence number of first actual message
+        this.initialClientSeqNum = -1;
```

Comment:
Can you please add assert somewhere that snapshot's sequence number is not above min reference sequence number across all messages in this.initialStates? That's required for correctness

Reply:
I believe this is covered by the error thrown in PendingStateManager.rebaseAt()